src/Core/Tool/Authorizer/PostAuthorizer.php

```diff
@@ -140,11 +135,28 @@ public function isAllowed(Entity $entity, $privilege)
             return true;
         }
 
-        // We check if the user is the owner of this post. If so, they are allowed
-        // to do almost anything, **except** change ownership and status of the post, which
-        // only admins can do.
-        if ($this->isUserOwner($entity, $user) && !$entity->hasChanged('user_id')
-            && $privilege !== 'change_status') {
+        // Only admins or users with 'Manage Posts' permission can change status
+        if ($privilege === 'change_status') {
+            return false;
+        }
+
+        // Only admins or users with 'Manage Posts' permission can the ownership of a post
```

Comment:
missing the word changed should be
// Only admins or users with 'Manage Posts' permission can change the ownership of a post